File: prototypeLeger/app/src/main/java/com/example/scrabbleprototype/model/Room.kt
package com.example.scrabbleprototype.model
import com.fasterxml.jackson.annotation.JsonIgnore
import kotlinx.serialization.Serializable

enum class State {
    Playing,
    Waiting,
    Finish,
}

@Serializable
class Room(var id: String, var socketIds: ArrayList<String>, var gameSettings: GameSettings, var state: State,
           var aiPlayersNumber: Int, var humanPlayersNumber: Int, var observers: ArrayList<User>
) {}

File: client/src/app/classes/room.ts
import { GameSettings } from '@common/game-settings';
export enum State {
    Playing,
    Waiting,
    Finish,
}

export class Room {
    id: string;
    gameSettings: GameSettings;
    state: State;
    socketIds: string[];
    aiPlayersNumber: number;
    humanPlayersNumber: number;

    constructor(id: string, gameSettings: GameSettings, state: State, socketIds: string[], aiNumber: number, humanNumber: number) {
        this.id = id;
        this.gameSettings = gameSettings;
        this.state = state;
        this.socketIds = socketIds;
        this.aiPlayersNumber = aiNumber;
        this.humanPlayersNumber = humanNumber;
    }
}
